File: 05_flutter_bank_base_with_framework/FlutterBankXml/app/src/main/java/kr/flutterbank/aos/xml/AccountManager.kt
package kr.flutterbank.aos.xml

import androidx.lifecycle.MutableLiveData

object AccountManager {
  val accounts = MutableLiveData<List<Account>>(arrayListOf(
    Account("10011"),
    Account("20012")
  ))

  fun checkAccountNoValid(accountNo: String): Boolean {
    val accts = accounts.value.orEmpty()
    return accountNo.isNotBlank() && !accts.map { it.no }.contains(accountNo)
  }

  fun makeAccount(accountNo: String): Boolean {
    val accts = accounts.value.orEmpty().toMutableList()
    accts.add(Account(accountNo))
    accounts.postValue(accts)
    return true
  }
}

File: 06_flutter_bank_base_with_flutter_boost/FlutterBankUIKit/FlutterBankUIKit/AccountManager.swift
//
//  AccountManager.swift
//  FlutterBankUIKit
//
//  Created by sjkim on 3/3/24.
//

class AccountManager {
  static let shared = AccountManager()
  
  var accounts: [Account] = [
    Account(no: "10011"),
    Account(no: "20012")
  ]
  
  func checkAccountNoValid(_ accountNo: String) -> Bool {
    if !accountNo.isEmpty && !accounts.map({ $0.no }).contains(accountNo) {
      return true
    }
    return false
  }
  
  func addAccount(_ accountNo: String) -> Bool {
    accounts.append(Account(no: accountNo))
    return true
  }
}

